web/packages/core/src/shadow-template.ts

```diff
@@ -218,13 +222,91 @@ ruffleShadowTemplate.innerHTML = `
             border-bottom: 1px solid lightgray;
             margin: 2px;
         }
+
+        #preloader {
+            display: flex;
```

Comment:
In my opinion, it looks better when the preloader covers the entire space of the player. This can be done by adding `#preloader` to the following list:
https://github.com/ruffle-rs/ruffle/blob/41bfc028d2b67abbba3c7285ae54b5a9652aa853/web/packages/core/src/shadow-template.ts#L33-L39

Then, after removing some redundant CSS properties, I ended up with just these working:
```css
#preloader {
    display: flex;
    flex-direction: column;
    background: var(--preloader-background, var(--ruffle-blue));
    align-items: center;
    justify-content: center;
}
```

Reply:
This was the original preloader design, but in #6839, Mike said "Get rid of the blue background. This works good for games, but it's not good for content like Flash web sites, which really want to display the native background color of the embed.... Using a transparent BG is a bit ugly, though. Maybe we can use a smaller box instead of taking over the entire container."

Maybe it's not as big a deal now that --preloader-background exists though, but sites using unpkg may not realize they need to change things.

Reply:
Also, without the aspect-ratio, the size of things like the Ruffle logo and loading bar and spinner can really vary depending on the size of the animation. For example, think of the header and footer on this page: http://www.dnai.org/timeline/.

Reply:
Though taking over the entire container seems fine to me, I think that if we want a smaller box, then the canvas must be hidden as well, otherwise the demo page looks really weird when switching between SWFs. To do so I suggest editing `hidePreloader()` and `showPreloader()` as follows:
```ts
private hidePreloader(): void {
    this.preloader.classList.add("hidden");
    this.container.classList.remove("hidden");
}

private showPreloader(): void {
    this.preloader.classList.remove("hidden");
    this.container.classList.add("hidden");
}
```

Reply:
> otherwise the demo page looks really weird when switching between SWFs

The demo page unfortunately doesn't show the preloader when switching between SWFs. This is because set_metadata runs nearly instantaneously when switching SWFs, even when the new SWF takes a long time to load. I'm unsure why this might be.

Reply:
Also, the preloader is part of the container as it is now.

Reply:
http://www.dnai.org/timeline/index.html

Version that takes up part of the screen:
![Screenshot from 2022-07-19 09-36-23](https://user-images.githubusercontent.com/29206584/179767531-3df47b38-ecd4-4582-8e4f-0bd3bb2ce613.png)

Version that takes up the whole screen:
![Screenshot from 2022-07-19 09-40-09](https://user-images.githubusercontent.com/29206584/179767532-6214ff48-de88-40f2-b065-0784afb1950f.png)

Page after loading:
![Screenshot from 2022-07-19 09-40-16](https://user-images.githubusercontent.com/29206584/179767535-e2bccaae-2410-463f-ad8a-b33411090f50.png)

Reply:
Switched to taking over the whole container.